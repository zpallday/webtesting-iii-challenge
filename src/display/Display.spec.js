import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Display from './Display';


describe('<Display />', () => {

    it('matches snapshot', () => {
        const tree = renderer.create(<Display />);
    
 
        expect(tree.toJSON()).toMatchSnapshot();
      })


it('Should run red-led locked', () => {
    const component = render(<Display closed={true} locked={true}/> )
    const display = component.getByText('Locked')
    expect(display)
})

it('Should run green-led unlocked', () => {
    const component = render(<Display closed={true} locked={false}/> )
    const displaygreen = component.getByText('Unlocked')
    expect(displaygreen)
})

it('should display unlocked while unlocked', () => {
    const component = render(<Display closed={true} locked={false} />)
    component.getByText('Unlocked')
})

it('should display red when closed', () => {
    const component =render(<Display closed={true} locked={false} /> )
    const displayclosed = component.getByText('Closed')
    expect(displayclosed)
})

it('Should open while green-led', () => {
    const component = render(<Display closed={false} locked={false} /> )
    const displayopen = component.getByText('Open')
    expect(displayopen)
})


})