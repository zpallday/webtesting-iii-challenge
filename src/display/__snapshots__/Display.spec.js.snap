// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
